src/prefect/settings.py

```diff
@@ -461,9 +476,64 @@ def valid_setting_names(cls) -> Set[str]:
             ):
                 settings_fields.update(field.annotation.valid_setting_names())
             else:
-                settings_fields.add(
-                    f"{cls.model_config.get('env_prefix')}{field_name.upper()}"
+                if field.validation_alias and isinstance(
+                    field.validation_alias, AliasChoices
+                ):
+                    for alias in field.validation_alias.choices:
+                        if not isinstance(alias, str):
+                            continue
+                        settings_fields.add(alias.upper())
+                else:
+                    settings_fields.add(
+                        f"{cls.model_config.get('env_prefix')}{field_name.upper()}"
+                    )
+        return settings_fields
+
+    @classmethod
+    def _get_settings_fields(
+        cls, accessor_prefix: Optional[str] = None
+    ) -> Dict[str, Any]:
```

Comment:
are we able to be more specific here? even with the recursion, we should always be returning 
```suggestion
    ) -> Dict[str, Setting]:
```
right?